### ISSUE TITLE
Resampling can not be debugged with error=recover

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlr3#611]
It would be nice if debugging using `options(error=recover)` could work when doing resampling. However, because `future_lapply()` is used, this is not possible, even when parallelization is not even used.
```r
> options(error=recover)
> l <- lrn("classif.debug", error_train = 1)
> resample(tsk("iris"), l, rsmp("holdout"))
INFO  [01:20:24.014] [mlr3]  Applying learner 'classif.debug' on task 'iris' (iter 1/1) 
Error in .__LearnerClassifDebug__.train(self = self, private = private,  : 
  Error from classif.debug->train()

Enter a frame number, or 0 to exit   

1: resample(tsk("iris"), l, rsmp("holdout"))
2: future.apply::future_lapply(seq_len(n), workhorse, task = task, learner = l
3: future_xapply(FUN = FUN, nX = nX, chunk_args = X, args = list(...), get_chu
4: value(fs)
5: value.list(fs)
6: resolve(y, result = TRUE, stdout = stdout, signal = signal, force = TRUE)
7: resolve.list(y, result = TRUE, stdout = stdout, signal = signal, force = TR
8: signalConditionsASAP(obj, resignal = FALSE, pos = ii)
9: signalConditions(obj, exclude = getOption("future.relay.immediate", "immedi
```

This is because the code in `future` runs all code first and inspects it for errors later. Could we drop in vanilla `lapply()` in our `resample()` (and in other places where `future_lapply()` is being used) whenever parallelization is not used? We could even drop our dependency on the `future.lapply` package and turn it into a suggested package like this.

**Comment by mllg** [comment]
> This is because the code in `future` runs all code first and inspects it for errors later. Could we drop in vanilla `lapply()` in our `resample()` (and in other places where `future_lapply()` is being used) whenever parallelization is not used? We could even drop our dependency on the `future.lapply` package and turn it into a suggested package like this.

Actually, this was implemented before mlr3-0.3.0, and future was only suggested. I've changed this to let future handle the RNG, regardless of the backend. I can however provide a debug option for debugging to call lapply instead of future.

**Comment by mb706** [comment]
Would be nice to have the option to `lapply`. Although if it is just seeds we could also sample a seed for each iteration that is then set in the `workhorse` function (and reset rng state in `on.exit()`).
